### PR TITLE
More accurate pagination URLs for Reader, Group, and Author pages

### PR DESCRIPTION
**Pull request by garethsime**
Before this change, the pagination URLs just pointed to `#`, which meant that you couldn't do things like 'Open in new tab', or bookmark a specific page if you were sorting by Release Date.

This commit helps solve those problems on the Reader, Group, and Author pages by:

 * Specifying a better URL for the pagination links (solves for 'Open in new tab')
 * Pushing the `search_order` parameter into the page URL when new pages are fetched from the server (solves for accurate bookmarking)

The changes are more or less the same for each of the three pages, except groups don't have a sort order.

(I haven't fixed it on the Search page, it's got a bit more going on in terms of query params etc.)

### CODE REVIEW

File: application/views/catalog/partials/footer.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ function get_search_page_from_url() {
 	var search_page = get_search_page_from_url();
 	set_advanced_form_page(search_page);
 
-	var search_order = 'alpha';
+	var search_order = $('.js-sort-menu').val() || 'alpha';
```
Comment (garethsime) [inline application/views/catalog/partials/footer.php:48]: `.js-sort-menu` is the sort order `<select>`. We're already going to some pains to keep it in sync with `search_order`, so I figured it wouldn't be the worst thing if I initialized this variable based on that (if it's available)

File: application/controllers/catalog/Author.php [hunk 2]
```diff
@@ -21,6 +21,8 @@ public function index($author_id)
 		$matches = $this->_get_all_author($author_id);
 		$this->data['matches'] = count($matches);
 
+		$this->data['search_order'] = $this->input->get('search_order');
```
Comment (garethsime) [inline application/controllers/catalog/Author.php:24]: This gets used to mark the correct sort option as selected in `views/catalog/author.php`

**Comment by redrun45** [comment]
I've tested this with the latest master, and it seems to do exactly what it says on the tin.
- Making the search order accessible in this way makes perfect sense to me
- I'm not qualified to review the new function parameter to the core pagination, but I do like how it's documented.

I've kicked the tires a bit, and nothing crashes when I give unexpected values in the URL
- Negative or zero page numbers are reset to 1
- Too-large page numbers show no results, but have controls for returning to page 1
- Unknown values for search_order are reset to 'alpha'

**Comment by redrun45** [comment]
This looks like a good change to me, but I'll be checking with other admins before I say we're ready for it.  So: "pending consensus".  😉

**Comment by garethsime** [comment]
Awesome, thanks 🙂 I'll be away for a few days starting tomorrow, so no rush since I won't be near my computer anyway

**Comment by notartom** [comment]
@redrun45 want me to deploy this on staging to let other admins play around? I still haven't figured out a sane way to make staging multi-user, and allow both of us to "demo" PRs so to speak, so for now we're stuck with just me doing as you direct :)

**Comment by redrun45** [comment]
@notartom that sounds great!  I'm not so much directing as giving a little push, then letting things take the path of least resistance.  :laughing: 

So yes, if this is the next one you'd like to test, let's do it.  I'll show the other admins what's different, ask them to try it out, and get any feedback we can.  It looks like there are some merge conflicts since my last review, but I can _try_ to get those resolved either today, or tomorrow evening, if that would be helpful to you.

**Comment by garethsime** [comment]
@redrun45 I've pushed a merge to fix the conflicts :slightly_smiling_face:

**Comment by notartom** [comment]
Sweet, I've checked our this PR on [staging](https://forum.librivox.org/viewtopic.php?t=69859).

**Comment by redrun45** [comment]
Alright!  I think we'll let things settle after the update tomorrow, and then I'll point folks there to try it out on their own timing.

**Comment by redrun45** [comment]
I updated the timeline on this one in the admin thread: it was a bit much to ask folks to try it out, along with the new changes, without a weekend to do it in.

**Comment by redrun45** [comment]
After two weeks, I think we've tested this as much as there is interest in doing so.  No strange behaviors I could find, this does just what it should.  Admin consensus: :+1:

**Comment by notartom** [comment]
@garethsime looks like this needs a rebase

**Comment by notartom** [comment]
I took the liberty of rebasing this myself, and proposed https://github.com/LibriVox/librivox-catalog/pull/215

**Comment by notartom** [comment]
Done in https://github.com/LibriVox/librivox-catalog/pull/215

**Comment by garethsime** [comment]
Awesome, thanks all!

**Pull request closed without merging**